### PR TITLE
fix: persist the replies when switching conversations - AP-235

### PR DESCRIPTION
**Pull request by davidchan1219**
**What this PR does** 📖

**Which issue(s) this PR fixes** 🔨

Fixes # AP-235

**Special notes for reviewers** 🗒️

**Additional comments** 🎤

### CODE REVIEW

**Review by stavares843 — CHANGES_REQUESTED** [review]
fixed but this is introducing a bug that doesn't happen on dev


it doesn't allow sending out emojis


https://user-images.githubusercontent.com/29093946/145878678-a7ad28dd-ba3c-434f-ae3e-f818adfdaf50.mp4

**Review by stavares843 — CHANGES_REQUESTED** [review]
always getting this error going to any DM on this branch

Captura de ecrã 2021-12-14, às 17 25 20

File: components/tailored/core/chatbar/Chatbar.vue [hunk 1]
```diff
@@ -51,6 +51,13 @@ export default Vue.extend({
       type: Object as PropType<Friend>,
     },
   },
+  // updated() {
+  //   const chatMsgList = localStorage.getItem('chat-message-list')
+  //   let cMLObj = chatMsgList ? JSON.parse(chatMsgList) : []
+  //   if (this.$props.recipient && cMLObj.some((e: any) => e.id === this.$props.recipient.address)) {
+  //     this.value = cMLObj.find((item: any) => item.id === this.$props.recipient.address).value
+  //   }
+  // },
```
Comment (stavares843) [inline components/tailored/core/chatbar/Chatbar.vue:60]: we can burn this commented out code

Comment (davidchan1219) [reply components/tailored/core/chatbar/Chatbar.vue]: Removed that commented part

**Comment by RetroPronghorn** [comment]
@davidchan1219 This approach is a bit too complex for what we need to acheive. Instead let's follow a pattern like so.


Make a new state section for message replies (create a Chat folder in store)

add the correct typings for it like 

```
type ChatState = {
    replies: {
        User.ID: string
    },
    text: {
       User.ID: string
    }
}
```

Make a mutation to add text based on the user we're chatting with and allow us to choose if it's a reply or generic chatbar text.

before the component unmounts call the mutation to store the text

```
beforeUnmount() {
   ...if we have an active reply being typed
       ...call chat mutation to store in reply state
  ...if we have active message in chatbar
       ...call chat mutation to store in text state
 }
 ```
 now when the component mounts, we can simply check the state to see if a reply is being made. We should also clear the reply from state when remounting.
 
 ```
 mounted() {
 ...if store.state.chat.replies[user]
    ...this.$data.reply = store.state.chat.replies[user]
    ...call mutation to clear
 ...if store.state.chat.text[user]
    ...this.store.state.chatbarContent = this.store.state.chat.text[user]
    ...call mutation to clear
}
```

this is psuedo code but please feel free to ask questions if you have any. I think we should start fresh from the dev branch on this. It will be a better approach in the long run.

**Comment by stavares843** [comment]
the crash is fixed ✅ 

persisting message bug is fixed ✅ 

sending emojis works ✅

**Review by stavares843 — APPROVED** [review]

**Review by WanderingHogan — APPROVED** [review]

**Pull request merged**